examples/src/ray_traced_triangle/mod.rs

```diff
@@ -0,0 +1,459 @@
+use glam::{Mat4, Vec3};
+use std::time::Instant;
+use std::{env, mem};
+use wgpu::ray_tracing::{
+    AccelerationStructureFlags, AccelerationStructureUpdateMode, BlasBuildEntry, BlasGeometries,
+    BlasGeometrySizeDescriptors, BlasTriangleGeometry, BlasTriangleGeometrySizeDescriptor,
+    CommandEncoderRayTracing, CreateBlasDescriptor, CreateTlasDescriptor, DeviceRayTracing,
+    TlasInstance, TlasPackage,
+};
+use wgpu::util::{BufferInitDescriptor, DeviceExt};
+use wgpu::{include_wgsl, BufferUsages, SamplerDescriptor};
+
+struct Example {
+    tlas_package: TlasPackage,
+    compute_pipeline: wgpu::ComputePipeline,
+    blit_pipeline: wgpu::RenderPipeline,
+    bind_group: wgpu::BindGroup,
+    blit_bind_group: wgpu::BindGroup,
+    storage_texture: wgpu::Texture,
+    start: Instant,
+}
+
+#[repr(C)]
+#[derive(bytemuck::Pod, bytemuck::Zeroable, Clone, Copy, Debug)]
+struct Uniforms {
+    view_inverse: [[f32; 4]; 4],
+    proj_inverse: [[f32; 4]; 4],
+}
+
+impl crate::framework::Example for Example {
+    fn required_features() -> wgpu::Features {
+        wgpu::Features::RAY_TRACING_ACCELERATION_STRUCTURE | wgpu::Features::RAY_QUERY
+    }
+
+    fn required_limits() -> wgpu::Limits {
+        wgpu::Limits::default()
+    }
+
+    fn init(
+        config: &wgpu::SurfaceConfiguration,
+        _adapter: &wgpu::Adapter,
+        device: &wgpu::Device,
+        queue: &wgpu::Queue,
+    ) -> Self {
+        let mut index_buffer = false;
```

Comment:
If this is forced it should be off because it's the only example that doesn't get compression from the index buffer, and so all other examples have index buffers, therefore if these are forced on we lose some testing coverage.

Reply:
My personal preference would be to have an index buffer in the example (to show the user how they work) and have a simple test which covers the non-index case.